src/plots/layout_attributes.js

```diff
@@ -188,6 +188,29 @@ module.exports = {
         editType: 'calc',
         description: 'Sets the default trace colors.'
     },
+    colorscale: {
```

Comment:
Let's put these new attributes and default logic in the [`Colorscale`](https://github.com/plotly/plotly.js/tree/master/src/components/colorscale) component module. 

Create two new files:

- src/components/colorscale/layout_attributes.js
- src/components/colorscale/layout_defaults.js

and link their `exports` in src/components/colorscale/index.js as `layoutAttributes` and `supplyLayoutDefaults`. The latter will get automatically picked in in `Plotly.supplyLayoutModuleDefaults` here:

https://github.com/plotly/plotly.js/blob/695f3114c86fee146c7f8e4b8700fa0b00d5c304/src/plots/plots.js#L1512-L1518